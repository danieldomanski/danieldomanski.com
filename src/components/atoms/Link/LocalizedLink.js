import React from 'react'
import { Link } from 'gatsby'
import { Location } from '@reach/router'
import styled from 'styled-components'
import tw from 'tailwind.macro'
import PropTypes from 'prop-types'
import { layout, typography, space, color } from 'styled-system'
import locales from '../../../config/locales'
import { formatUrlToLocale } from '../../../utils/format'

const LinkContainer = styled(Link)`
  ${tw`font-sans no-underline cursor-pointer`};
  ${space}
  ${layout}
  ${typography};
  ${color}

`

const LocalizedLink = ({
  children,
  fontSize,
  display,
  fontWeight,
  fontColor,
  width,
  maxWidth,
  my,
  partiallyActive,
  activeStyle,
  hover,
  to,
  ...rest
}) => (
  <Location>
    {({ location }) => {
      const isIndex = to === '/'
      const locale = formatUrlToLocale(location.pathname)

      const path = locales[locale].default
        ? to
        : `${locales[locale].path}${isIndex && to !== '/' ? '' : to}`

      return (
        <LinkContainer
          display={display}
          fontSize={fontSize}
          fontWeight={fontWeight}
          color={fontColor}
          width={width}
          maxWidth={maxWidth}
          my={my}
          activeStyle={activeStyle}
          partiallyActive={partiallyActive}
          to={path}
          style={{ transition: '0.3s padding' }}
          hover={hover}
          {...rest}
        >
          {children}
        </LinkContainer>
      )
    }}
  </Location>
)

LocalizedLink.propTypes = {
  fontSize: PropTypes.oneOf(['base', 'lg', 'xl', '2xl', '4xl', '5xl', '6xl']),
  children: PropTypes.node.isRequired,
  fontWeight: PropTypes.oneOf(['base', 'bold', 'black']),
  fontColor: PropTypes.string,
}

LocalizedLink.defaultProps = {
  fontSize: 'base',
  fontWeight: 'base',
  fontColor: 'primary.6',
}

export default LocalizedLink
