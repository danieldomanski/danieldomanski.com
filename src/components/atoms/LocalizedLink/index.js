import React from 'react'
import { Link } from 'gatsby'
import { Location } from '@reach/router'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { layout, typography, space, color } from 'styled-system'
import locales from '../../../config/locales'
import { formatUrlToLocale } from '../../../utils/format'

const LinkContainer = styled(Link)`
  ${space}
  ${layout}
  ${typography};
  ${color}

  text-decoration: none;
  cursor: pointer;
  
  &:hover {
    color: ${props => (props.hover ? props.hover.color : 'inherit')};
  }
`

const LocalizedLink = ({
  children,
  fontSize,
  display,
  fontWeight,
  fontColor,
  width,
  my,
  partiallyActive,
  activeStyle,
  hover,
  to,
  ...rest
}) => (
  <Location>
    {({ location }) => {
      const isIndex = to === '/'
      const locale = formatUrlToLocale(location.pathname)

      const path = locales[locale].default
        ? to
        : `${locales[locale].path}${isIndex && to !== '/' ? '' : to}`.replace(
            /\/+$/,
            ''
          )

      return (
        <LinkContainer
          display={display}
          fontSize={fontSize}
          fontWeight={fontWeight}
          color={fontColor}
          width={width}
          my={my}
          activeStyle={activeStyle}
          partiallyActive={partiallyActive}
          to={path}
          hover={hover}
          {...rest}
        >
          {children}
        </LinkContainer>
      )
    }}
  </Location>
)

LocalizedLink.propTypes = {
  fontSize: PropTypes.oneOf(['base', 'lg', 'xl', '2xl', '4xl', '5xl', '6xl']),
  children: PropTypes.node.isRequired,
  fontWeight: PropTypes.oneOf(['base', 'bold', 'black']),
  fontColor: PropTypes.string,
}

LocalizedLink.defaultProps = {
  fontFamily: 'sans',
  fontSize: 'base',
  fontWeight: 'base',
  fontColor: 'primary.6',
}

export default LocalizedLink
