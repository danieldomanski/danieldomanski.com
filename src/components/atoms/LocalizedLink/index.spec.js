import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { render, cleanup } from '@testing-library/react'

import LocalizedLink from './LocalizedLink.js'

afterEach(cleanup)

test('<Image> renders', () => {
  let mockData = {
    to: 'test',
    partiallyActive: true,
    display: 'inline-block',
  }

  const { container } = render(
    <LocalizedLink {...mockData}>test</LocalizedLink>
  )

  expect(container.firstChild).toMatchSnapshot()
})
