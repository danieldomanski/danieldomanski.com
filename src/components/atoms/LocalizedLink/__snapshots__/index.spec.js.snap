// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image> renders 1`] = `
.c0 {
  display: inline-block;
  font-size: base;
  font-weight: base;
  font-family: sans;
  color: primary.6;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  color: inherit;
}

<a
  class="c0"
  color="primary.6"
  display="inline-block"
  font-family="sans"
  font-size="base"
  font-weight="base"
  href="test"
>
  test
</a>
`;
